# HG changeset patch
# User Rik <rik@octave.org>
# Date 1433604381 25200
#      Sat Jun 06 08:26:21 2015 -0700
# Node ID b93a155dc2005418aba962123e9eaf53379b9c7c
# Parent  011a364b4d78a00d8ce279bdbb086745d678dfbb
ginput.m: Stop feval error when ginput called multiple times (bug #45261).

* ginput.m: Restore the original closereq function for the figure at the end of
a normal termination of ginput.

diff --git a/scripts/plot/util/ginput.m b/scripts/plot/util/ginput.m
--- a/scripts/plot/util/ginput.m
+++ b/scripts/plot/util/ginput.m
@@ -97,16 +97,17 @@ function varargout = ginput (n = -1)
       button = button(1:n);
     endif
 
   unwind_protect_cleanup
     if (isfigure (fig))
       ## Only execute if window still exists
       set (fig, "windowbuttondownfcn", orig_windowbuttondownfcn);
       set (fig, "keypressfcn", orig_keypressfcn);
+      set (fig, "closerequestfcn", orig_closerequestfcn);
     endif
   end_unwind_protect
 
   varargout = {x, y, button};
 
 endfunction
 
 function [x, y, n, button] = ginput_accumulator (mode, xn, yn, btn)
